impeller/scene/importer/importer_gltf.cc

```diff
@@ -2,20 +2,211 @@
+    {"COLOR_0", VerticesBuilder::Attribute::kColor},
+};
+
+static bool WithinRange(int index, size_t size) {
+  return index >= 0 && static_cast<size_t>(index) < size;
+}
+
+static bool ProcessStaticMesh(const tinygltf::Model& gltf,
+                              const tinygltf::Primitive& primitive,
+                              fb::StaticMeshT& static_mesh) {
+  //---------------------------------------------------------------------------
+  /// Vertices.
+  ///
+
+  {
+    VerticesBuilder builder;
+
+    for (const auto& attribute : primitive.attributes) {
+      auto attribute_type = kAttributes.find(attribute.first);
+      if (attribute_type == kAttributes.end()) {
+        std::cerr << "Vertex attribute \"" << attribute.first
+                  << "\" not supported." << std::endl;
+        continue;
+      }
+
+      const auto accessor = gltf.accessors[attribute.second];
+      const auto view = gltf.bufferViews[accessor.bufferView];
+
+      const auto buffer = gltf.buffers[view.buffer];
+      const unsigned char* source_start = &buffer.data[view.byteOffset];
+
+      VerticesBuilder::ComponentType type;
+      switch (accessor.componentType) {
+        case TINYGLTF_COMPONENT_TYPE_BYTE:
+          type = VerticesBuilder::ComponentType::kSignedByte;
+          break;
+        case TINYGLTF_COMPONENT_TYPE_UNSIGNED_BYTE:
+          type = VerticesBuilder::ComponentType::kUnsignedByte;
+          break;
+        case TINYGLTF_COMPONENT_TYPE_SHORT:
+          type = VerticesBuilder::ComponentType::kSignedShort;
+          break;
+        case TINYGLTF_COMPONENT_TYPE_UNSIGNED_SHORT:
+          type = VerticesBuilder::ComponentType::kUnsignedShort;
+          break;
+        case TINYGLTF_COMPONENT_TYPE_INT:
+          type = VerticesBuilder::ComponentType::kSignedInt;
+          break;
+        case TINYGLTF_COMPONENT_TYPE_UNSIGNED_INT:
+          type = VerticesBuilder::ComponentType::kUnsignedInt;
+          break;
+        case TINYGLTF_COMPONENT_TYPE_FLOAT:
+          type = VerticesBuilder::ComponentType::kFloat;
+          break;
+        default:
+          std::cerr << "Skipping attribute \"" << attribute.first
+                    << "\" due to invalid component type." << std::endl;
+          continue;
+      }
+
+      builder.SetAttributeFromBuffer(attribute_type->second,  // attribute
+                                     type,                    // component_type
+                                     source_start,            // buffer_start
+                                     accessor.ByteStride(view),  // stride_bytes
+                                     accessor.count);            // count
+    }
+
+    builder.WriteFBVertices(static_mesh.vertices);
+  }
+
+  //---------------------------------------------------------------------------
+  /// Indices.
+  ///
+
+  if (!WithinRange(primitive.indices, gltf.accessors.size())) {
+    std::cout << "Mesh primitive has no index buffer. Skipping." << std::endl;
```

Comment:
Noticing that other output in this file is going to stderr, but this one is going to stdout. Should it also be stderr?

Reply:
Yes, fixed